contracts/Synthswap.sol

```diff
@@ -21,28 +25,117 @@ contract SynthSwap is ISynthSwap {
     }
 
     function swapInto(
-        bytes calldata payload,
-        bytes32 destinationSynthCurrencyKey
+        address inputTokenAddress,
+        uint256 inputTokenAmount,
+        bytes32 destinationSynthCurrencyKey,
+        uint256 slippage
     ) external payable override returns (uint) {
-        // Make sure to set destReceiver to this contract
-        (bool success, bytes memory returnData) = aggregationRouterV3.delegatecall(payload);
-        require(success, _getRevertMsg(returnData));
-        require(false, "stop");
+
+        // Approve the 1inch router to spend sUSD.
+        IERC20 InputERC20 = IERC20(inputTokenAddress);
+        uint tokenBalance = InputERC20.balanceOf(address(this));
```

Comment:
We should think about how we want to handle a swap of ETH for sUSD.

The 1Inch address for ETH is `0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE`, but it won't be an ERC20.

Maybe we wrap it first if someone calls `swapIntoWithETH()` or something?

Reply:
Yes, I think that is a good idea!